### PR TITLE
Controlled Field component and updates to TapToEdit

### PR DESCRIPTION
**Pull request by kjkurtz**

### CODE REVIEW

File: packages/ui/src/Field.tsx [hunk 1]
```diff
@@ -36,58 +30,46 @@ export interface FieldProps extends FieldWithLabelsProps {
...
+  helperTextColor,
+}: FieldProps) => {
   // showCustomInput, customValue, and selectValue are all for type="customSelect"
   const [showCustomInput, setShowCustomInput] = useState(false);
   const [customValue, setCustomValue] = useState("");
-  const [selectValue, setSelectValue] = useState(props.initialValue || "");
-
-  const handleChange = (newValue: string) => {
-    if (props.type === "currency") {
-      newValue = newValue.replace("$", "");
```
Comment (joshgachnang) [inline packages/ui/src/Field.tsx:65]: I think we'll still want this to handle currency/percent fields without the caller having to deal with this.

File: packages/ui/src/Field.tsx [hunk 2]
```diff
@@ -36,58 +30,46 @@ export interface FieldProps extends FieldWithLabelsProps {
...
-
+export const Field = ({
+  name,
+  label,
+  labelColor,
+  height,
+  type,
+  rows,
+  value,
+  setValue,
```
Comment (joshgachnang) [inline packages/ui/src/Field.tsx:53]: I'd suggest something like onChange etc to match most other components in RN.

**Review by joshgachnang — COMMENTED** [review]
This looks on the right path!!

File: packages/ui/src/Field.tsx (hunk 1 above)
Comment (kjkurtz) [reply packages/ui/src/Field.tsx:65]: Agreed! I'll get that back in here shortly.

File: packages/ui/src/Field.tsx (hunk 2 above)
Comment (kjkurtz) [reply packages/ui/src/Field.tsx:53]: Seems reasonable.

File: packages/ui/src/Field.tsx (hunk 1 above)
Comment (joshgachnang) [reply packages/ui/src/Field.tsx:65]: Good deal! Wasn't sure if it was permanent or temporary while you're working it out, just wanted to make sure!

Comment (kjkurtz) [reply packages/ui/src/Field.tsx:65]: For sure, a good reminder! I was just making sure that as I make the updates within our primary app that the changes required are reasonable before polishing it up. I'll request a new review here once I'm good there. Thanks for the feedback!

File: packages/ui/src/Field.tsx [hunk 3]
```diff
@@ -13,13 +13,7 @@ import {TextField} from "./TextField";
 export interface FieldProps extends FieldWithLabelsProps {
   name: string;
   label?: string;
-  subLabel?: string;
-  initialValue?: any;
-  handleChange: any;
   height?: number;
-  // Additional validation
-  validate?: (value: any) => boolean;
```
Comment (kjkurtz) [inline packages/ui/src/Field.tsx:21]: I can't find anywhere that actually uses these validate props so I removed them. If there is a spot I missed, let me know!

File: packages/ui/src/Field.tsx (hunk 1 above)
Comment (kjkurtz) [reply packages/ui/src/Field.tsx:65]: This should be there now. It's in a different spot (right near the text component) but should do the same thing!

File: packages/ui/src/Field.tsx (hunk 3 above)
Comment (joshgachnang) [reply packages/ui/src/Field.tsx:21]: Ah yeah, not sure this actually got used once Field had things like decimals implemented. 👍

File: packages/ui-demo/src/Field.stories.tsx [hunk 4]
```diff
@@ -12,12 +12,12 @@ export const FieldStories = {
...
         <StorybookContainer>
           <Field
-            handleChange={() => {}}
             helperText="Here's some help text"
-            initialValue="Pre-filled text"
             label="Text Field"
             name="text"
             type="text"
+            value="Pre-filled text"
+            onChange={() => {}}
```
[the remark this answers is not in the document]
Comment (kjkurtz) [reply packages/ui-demo/src/Field.stories.tsx:20]: 😂 yes. Appears I pushed up flourish but not this. I'll take care of it when I'm back at my laptop.

File: packages/ui/src/Field.tsx [hunk 5]
```diff
@@ -36,58 +30,46 @@ export interface FieldProps extends FieldWithLabelsProps {
...
 
   // Custom select has 3 values - the overall field value, the value of the select menu, and the value of the custom input
   const handleCustomSelectListChange = (newValue: string) => {
     // If "custom" is selected from the dropdown, toggle the custom input open and clear the previous value
     if (newValue === "custom") {
       setShowCustomInput(true);
-      setValue("");
-      setSelectValue(newValue);
+      onChange("");
+      // setSelectValue(newValue);
```
Comment (swanka) [inline packages/ui/src/Field.tsx:72]: Do we still need this comment in here?

Comment (kjkurtz) [reply packages/ui/src/Field.tsx]: Nope, gone now!

**Review by swanka — APPROVED** [review]
This is an awesome change

**Pull request merged**